app/src/context/SearchContext.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+} from "react";
+import { usePathname, useSearchParams } from "next/navigation";
+import {
+  DEFAULT_PAGE_NUM,
+  DEFAULT_SORT,
+  DEFAULT_SEARCH_TERM,
+} from "../config/constants";
+import { Filter, SearchManager } from "../utils/searchManager";
+
+interface SearchContextType {
+  searchManager: SearchManager;
+}
+
+export const stringToFilter = (filtersString: string | null): Filter[] => {
+  if (!filtersString) return [];
+
+  const res = filtersString
+    .split(",")
+    .map((filterPair) => {
+      //??
+      const match = filterPair.match(/^\[(\w+)=(.+)\]$/);
+      if (match) {
+        const [, filter, value] = match;
+        return { filter, value };
+      }
+      return null;
+    })
+    .filter((filter): filter is Filter => filter !== null);
+  return res;
+};
+
+export const filterToString = (filters: Filter[]): string => {
+  if (!filters || filters.length === 0) return "";
+
+  return filters.map(({ filter, value }) => `[${filter}=${value}]`).join(",");
+};
+
+const SearchContext = createContext<SearchContextType | undefined>(undefined);
+
+export const SearchProvider = ({ children }: { children: React.ReactNode }) => {
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  // const searchManager = useMemo(() => {
+  //   const params = Object.fromEntries(searchParams.entries());
+
+  //   return new SearchManager({
+  //     initialPage: Number(params.page) || DEFAULT_PAGE_NUM,
+  //     initialSort: params.sort || DEFAULT_SORT,
+  //     initialFilters: stringToFilter(params.filters),
+  //     initialKeywords: params.keywords || DEFAULT_SEARCH_TERM,
+  //     isCollectionSearch: params.isCollectionSearch === "true",
+  //   });
+  // }, [pathname, searchParams]);
+
+  const searchManagerRef = useRef<SearchManager>(
+    new SearchManager({
+      initialPage: DEFAULT_PAGE_NUM,
+      initialSort: DEFAULT_SORT,
+      initialFilters: [],
+      initialKeywords: DEFAULT_SEARCH_TERM,
+      isCollectionSearch: false,
+    })
+  );
+
+  useEffect(() => {
```

Comment:
I don't necessarily agree with this. I removed `useEffect`s in this PR in Turbine https://github.com/NYPL/nypl-ds-test-app/pull/122/files#diff-3af8a06a249633ad6569f63b96449a1b73fe6e3e58dca0396e748ae36234443e

What you have going on this is workflow is great and try to make it work without this `useEffect`.